Panic when unmarshalling devnet allocs file

<!--
Need help?
Refer to our contributing guidelines for additional information about making a good issue:
https://github.com/ethereum-optimism/.github/blob/master/CONTRIBUTING.md
-->

**Describe the bug**
In the current develop branch([`9a59811f51`](https://github.com/ethereum-optimism/optimism/tree/9a59811f53960e0fae39a1c7cbcc25c3dfc808e6)), the type of `nonce` is string hexadecimal in the generated devnet l1 alloc file (`.devnet/allocs-l1.json`). It makes panic when trying to run any e2e test, with the following message.
```
panic: cannot unmarshal dump: json: cannot unmarshal string into Go struct field DumpAccount.accounts.nonce of type uint64

goroutine 1 [running]:
github.com/ethereum-optimism/optimism/op-e2e/config.init.0()
	/Users/tei/tip-optimism/op-e2e/config/init.go:93 +0x730
exit status 2
```

The content of the `.devnet/allocs-l1.json`:
```
...
  "accounts": {
    "0x2abb7cbb720020ee3c9ecf3915d14b6d1886a577": {
      "nonce": "0x2",
      "balance": "0x0",
      "code": "0x60806040523...",
      "storage": {
        "0x0000000000000000000000000000000000000000000000000000000000000000": "0x0000000000000000000000006153107879eb0f3743b33091206fd2946833c35b"
      }
    },
...
```

**To Reproduce**
Steps to reproduce the behavior:
1. Checkout the latest develop branch (`9a59811f51`)
2. Rebuild everything (following `CONTRIBUTING.md`)
3. Generate devnet alloc file (following `op-e2e/README.md`)
4. Run e2e test (e.g. `go test -run TestOutputCannonGame` in `op-e2e/faultproofs`)

**Expected behavior**
The type of nonce must be an integer, or the code can unmarshal hex string.

**System Specs:**
 - OS: macOS Sonoma 14.3 (M1 pro)
 - Package Version (or commit hash): `9a59811f51`

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2.1
 parameters:
   ci_builder_image:
     type: string
-    default: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:v0.40.0
+    default: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:v0.41.0
   base_image:
     type: string
     default: ubuntu-2204:2022.07.1

--- a/op-chain-ops/genesis/config.go
+++ b/op-chain-ops/genesis/config.go
@@ -745,7 +745,7 @@ func NewStateDump(path string) (*gstate.Dump, error) {
 // It appears as if updates to foundry have changed the serialization of the state dump.
 type ForgeDump gstate.Dump
 
-func (d *ForgeDump) UmashalJSON(b []byte) error {
+func (d *ForgeDump) UnmarshalJSON(b []byte) error {
 	type forgeDumpAccount struct {
 		Balance   string                 `json:"balance"`
 		Nonce     hexutil.Uint64         `json:"nonce"`

--- a/op-chain-ops/genesis/config.go
+++ b/op-chain-ops/genesis/config.go
@@ -733,11 +733,53 @@ func NewStateDump(path string) (*gstate.Dump, error) {
 		return nil, fmt.Errorf("dump at %s not found: %w", path, err)
 	}
 
-	var dump gstate.Dump
-	if err := json.Unmarshal(file, &dump); err != nil {
+	var fdump ForgeDump
+	if err := json.Unmarshal(file, &fdump); err != nil {
 		return nil, fmt.Errorf("cannot unmarshal dump: %w", err)
 	}
+	dump := (gstate.Dump)(fdump)
 	return &dump, nil
+}
+
+// ForgeDump is a simple alias for state.Dump that can read "nonce" as a hex string.
+// It appears as if updates to foundry have changed the serialization of the state dump.
+type ForgeDump gstate.Dump
+
+func (d *ForgeDump) UmashalJSON(b []byte) error {
+	type forgeDumpAccount struct {
+		Balance   string                 `json:"balance"`
+		Nonce     hexutil.Uint64         `json:"nonce"`
+		Root      hexutil.Bytes          `json:"root"`
+		CodeHash  hexutil.Bytes          `json:"codeHash"`
+		Code      hexutil.Bytes          `json:"code,omitempty"`
+		Storage   map[common.Hash]string `json:"storage,omitempty"`
+		Address   *common.Address        `json:"address,omitempty"`
+		SecureKey hexutil.Bytes          `json:"key,omitempty"`
+	}
+	type forgeDump struct {
+		Root     string                              `json:"root"`
+		Accounts map[common.Address]forgeDumpAccount `json:"accounts"`
+	}
+	var dump forgeDump
+	if err := json.Unmarshal(b, &dump); err != nil {
+		return err
+	}
+
+	d.Root = dump.Root
+	d.Accounts = make(map[common.Address]gstate.DumpAccount)
+	for addr, acc := range dump.Accounts {
+		d.Accounts[addr] = gstate.DumpAccount{
+			Balance:   acc.Balance,
+			Nonce:     (uint64)(acc.Nonce),
+			Root:      acc.Root,
+			CodeHash:  acc.CodeHash,
+			Code:      acc.Code,
+			Storage:   acc.Storage,
+			Address:   acc.Address,
+			SecureKey: acc.SecureKey,
+		}
+	}
+	return nil
 }
 
 // NewL2ImmutableConfig will create an ImmutableConfig given an instance of a

